### PR TITLE
fixes knights

### PR DESCRIPTION
**Pull request by coffeecoffeecoffee**
## About The Pull Request
FIXES FOOTKNIGHTS AND MEN AT ARMS SPAWNING WITH STACKING LEGENDARY SKILLS.
Swaps a little bit of the armor between men at arms and foot knights, because - did you know chain coifs are actually better neck protection than gorgets? Why did men at arms, part of whose specialization is specifically in crossbows, get better armor than the knights?
Gives knights steel booties as it's kind of just a source of pointless tedium, but also nerfs their armor SOMEWHAT. Before, they used the "steel cuirass", which was actually a very high-durability medium-weight armor which protects the chest - now, MOST of them get "coat of arms", which are actually less durability, but makes proper use of knights' heavy armor trait, and protects slightly more.
Heavy knights get a proper bervor and half-plate to make up for the fact they're slow and the only knights which don't get shields, as two-handed weapon specialists.
Swapped heavy knight specialization in polearms for swords and axes, as polearms don't take high-strength and there are a lot of iconic heavy two-handed swords for them to use - as well, axes were always a two-handed weapon, which is supposed to be their specialty.
Swapped foot knight specialization in axes for swords, so they can fit the fantasy of the sword-and-board knight (or flail! they kept that!) and properly use their one-handed specializations. They got some shield skill to keep them distinct.
Mounted knights more or less got a buff so they can use spears and maces on their mounts, as these are traditionally damn good weapons to use on a horse, and because it keeps them balanced in weapon skills with the others.

TL;DR: fixes a critical bug, makes the armor knights get make sense, makes their main chest armor worse mostly but gives them a little more weapon skills to make up for it
No single knight loadout is so objectively the worst anymore
They are overall stronger, but still nowhere what they were on Ratwood

### CODE REVIEW

**Comment by coffeecoffeecoffee** [comment]
WAAAIIIT I didn't quite totally fix men at arms
